### DISCUSSION TITLE
How do I deal with constant fan spin-up and spin-down due to semi-passive fan configuration?

### DISCUSSION THREAD

**Discussion by kamild1996** [Rem0o/FanControl.Releases#1589] [Q&A] [answered]
I'm trying to figure out how to make it so the fans on my GPU aren't in a constant loop of revving up and down to 0 when I'm playing lighter games. Even with hysteresis set to like 5 degrees and increased response time I cannot make it stop doing that - these settings just seem to make it that this problem occurs less often, but they create another problem where the fan speeds fluctuate a lot when the card is under proper load in more demanding games. Changing the zero RPM temperature threshold doesn't seem to help, the same thing still happens, just on lower/higher temps.

My current fan curve (on this particular GPU, 40% is the lowest fan which makes the fans spin up): 
![image](https://user-images.githubusercontent.com/5320414/222923271-4047a2e3-31d6-4e8e-95bb-9b9e5cd3b63d.png)

I wouldn't mind disabling this whole mechanic altogether since at the lowest possible fan speed I don't mind their subtle hum, but if this configuration has the purpose of increasing the lifespan of fans, I'd rather try to keep it on. But also I'm not sure how their lifespan changes with this problem going on, and the constant clicking when the fans start and stop is sufficiently annoying.

Is there a way I can setup Fan Control to prevent this from happening?

**Comment by Rem0o** [comment]
Hysteresis is ignored at the limits of your curve.   If you want hysteresis to apply at these limits, simply add another point like at 30 deg still at 0%, and an other at like 100 deg at 80%.

**Reply by Rem0o** [reply]
Goal of this behavior is for your fans to be able to reach the idle state. If you got a 10 deg hysteresis and your last change was at 51 degree for example, you'll need a 10 degree drop, so 41 degree to get to 0 rpm, which is not ideal.

If you get that start/stop behavior, your idle point is probably badly placed relative to where a light load is.

**Reply by kamild1996** [reply]
Thanks for the information about hysteresis. I did try to add a second point at the left side of the curve, and while it did make the hysteresis work, it greatly improved the situation, but didn't completely remove the problem - it does happen far less often.

As far as the idle point goes, that's the problem - I cannot find that point. One of my games is so light on the GPU, I can put the 0% point at 40-45 degrees and it will not only keep up the behavior in-game, it will also make it happen outside of gaming, just using my browser. At 50, it happens just in the game.

**Comment by JesusPwnsYa** [comment, marked as answer]
What I wouild recommend is using an average sensor for your curve. At least that's how I do it.
Just add a "Time average" custom sensor, choose the real sensor as a source and use this custom sensor in your curve.
I've set my gpu up with a 5 sec average, you could raise it to 10-15 seconds according to your needs.

Added later:
What I also did was creating a seperate curve for how it handles a lowering temperature. So it takes a lower temperature to turn the fan off, or a higher temperature to bring it back up to speed, with more control than with the hysteresis.

**Reply by kamild1996** [reply]
I'll try the time average sensor, thank you.
Would you mind elaborating a little bit on that separate curve thing? I'm having trouble implementing it in Fan Control.

**Reply by kamild1996** [reply]
Okay, I've had a second look and I think I see what you meant. Something like this, right?
![image](https://user-images.githubusercontent.com/5320414/224564487-e9f1d169-e7bc-482a-aa3a-6d8d4ea7b9df.png)
I'll try to test that soon and report back.

**Reply by Rem0o** [reply]
@kamild1996 yeah that's the general idea. You probably don't want your load fan speed to be a 100% though.  Play with the numbers.

**Reply by kamild1996** [reply]
> You probably don't want your load fan speed to be a 100% though. Play with the numbers.

My logic here was to make the load fan speed on the trigger 100%, so once the load temp is reached, the mix curve set to the "min" function gives control over the fans to the load curve. Seems to work as intended.

**Reply by Rem0o** [reply]
Ah! Didn't see the min, used to max for 99% of use cases 😅

**Reply by kamild1996** [reply]
Tested in-game, works great. With idle threshold set at 35C and load at 55C, fans start once after a moment and stay on for the duration of the GPU-lightweight game. Although sometimes when the GPU settles at 39C (lowest it goes in-game) and another game level is loading, the GPU can get enough downtime to sometimes hit 35C, causing the fans to stop. But that happens maybe once every 30-40 mins, so I'm fine with that. Also the average sensor is a nice way to smooth out temperature changes a little.
Thank you for the idea @JesusPwnsYa 😊
